apps/studio/components/interfaces/Settings/Addons/IPv4SidePanel.tsx

```diff
@@ -0,0 +1,284 @@
+                    <p className="text-sm">No IPv4</p>
+                  </div>
+                  <div className="px-4 py-2">
+                    <p className="text-foreground-light">
+                      Use connection pooler or IPv6 for direct connections
+                    </p>
+                    <div className="flex items-center space-x-1 mt-2">
+                      <p className="text-foreground text-sm">$0</p>
+                      <p className="text-foreground-light translate-y-[1px]"> / month</p>
+                    </div>
+                  </div>
+                </div>
+              </Radio>
+              {availableOptions.map((option) => (
+                <Radio
+                  className="col-span-4 !p-0"
+                  name="ipv4"
+                  key={option.identifier}
+                  disabled={isFreePlan}
+                  checked={selectedOption === option.identifier}
+                  label={option.name}
+                  value={option.identifier}
+                >
+                  <div className="w-full group">
+                    <div className="border-b border-default px-4 py-2 group-hover:border-control">
+                      <p className="text-sm">{option.name}</p>
+                    </div>
+                    <div className="px-4 py-2">
+                      <p className="text-foreground-light">
+                        Allow direct database connections via IPv4
+                      </p>
+                      <div className="flex items-center space-x-1 mt-2">
+                        <p className="text-foreground text-sm">${option.price}</p>
+                        <p className="text-foreground-light translate-y-[1px]"> / month</p>
+                      </div>
+                    </div>
+                  </div>
+                </Radio>
+              ))}
+            </Radio.Group>
+          </div>
+
+          {hasChanges && (
+            <>
+              {selectedOption === 'ipv4_none' ||
+              (selectedIPv4?.price ?? 0) < (subscriptionIpV4Option?.variant.price ?? 0) ? (
+                subscription?.billing_via_partner === false && (
+                  <p className="text-sm text-foreground-light">
+                    Upon clicking confirm, the add-on is removed immediately and any unused time in
+                    the current billing cycle is added as prorated credits to your organization and
+                    used in subsequent billing cycles.
+                  </p>
+                )
+              ) : (
+                <p className="text-sm text-foreground-light">
+                  Upon clicking confirm, the amount of{' '}
+                  <span className="text-foreground">${selectedIPv4?.price.toLocaleString()}</span>{' '}
+                  will be added to your monthly invoice.
+                </p>
+              )}
+            </>
+          )}
```

Comment:
**(note):** I'm pretty sure it's only me who's a bit confused here. But maybe you can double check whether the texts are correct (regarding "Prepaid", "End-of-Month", "Usage-Based", "Prorated Credits" etc.).